digest/utility.py

```diff
@@ -207,13 +198,17 @@ def count_unique_records(data: pd.DataFrame) -> int:
 
 def get_pipelines_overview(bagel: pd.DataFrame, schema: str) -> pd.DataFrame:
     """
-    Constructs a dataframe containing global statuses of pipelines in bagel.csv
-    (based on "pipeline_complete" column) for each participant and session.
+    Constructs a wide format dataframe from the long format input file,
+    with one row per participant-session pair and one column per event (e.g., pipeline, assessment)
     """
-    pipeline_complete_df = bagel.pivot(
+    # When there are duplicate entries for the combination of participant, session, & event, the first occurrence is kept
+    pipeline_complete_df = bagel.pivot_table(
         index=get_id_columns(bagel),
         columns=get_event_id_columns(bagel, schema),
         values=BAGEL_CONFIG[schema]["overview_col"],
+        aggfunc="first",
```

Comment:
1. cool
2. didn't know that was a df function (?)
3. says here this is going to be deprecated: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.first.html - not sure if this applies in the context here. Do you get deprecation warnings?

Reply:
Good question! I don't get any deprecation warnings on my end. I'm pretty sure this is because the function being used is not the dataframe one, but an aggregation method from pandas `groupby`: https://pandas.pydata.org/pandas-docs/stable/user_guide/groupby.html#built-in-aggregation-methods

It looks like [`pivot_table` uses `groupby` behind the hood](https://stackoverflow.com/a/72933069)